### ISSUE TITLE
Connect to externalDatabase using environment (Vault) variable

### ISSUE THREAD

**Issue by marekargalas** [airflow-helm/charts#528]
### Checks

- [X] I have checked for [existing issues](https://github.com/airflow-helm/charts/issues).
- [X] I have looked through the [FAQ](https://github.com/airflow-helm/charts/tree/main/charts/airflow#faq---airflow).

### Question

Hi, 

I am [injecting](https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar) Vault secrets to Airflow (using annotations not backend engine). I would like to also use it for getting PG database password to connect metastore. I [found](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html) how this could be done but I suppose just with running Vault backend. Is there a way to supply password from attached folder or environment variables? Thanks!

**Comment by thesuperzapper** [comment]
@marekargalas Right now the chart only supports getting the password from a Secret resource using `externalDatabase.passwordSecret` and `externalDatabase.passwordSecretKey`, there are a number of ways to automatically create Secrets from your Vault, for example:
- [secrets-store-csi-driver](https://secrets-store-csi-driver.sigs.k8s.io/topics/sync-as-kubernetes-secret.html)
- [external-secrets](https://github.com/external-secrets/external-secrets)

While it would be possible to create a value like `externalDatabase.passwordFile`, that specifies the path of a file in the pod which contains the password, I'm not sure how important that feature is, given the secret approach already exists.

___PS:__ we also plan to add `externalDatabase.userSecret` and `externalDatabase.userSecretKey` to support getting the username from a Secret also, which will probably make vault integration even easier._

**Comment by thesuperzapper** [comment]
> @thesuperzapper thanks for clarification! So `AIRFLOW__CORE__SQL_ALCHEMY_CONN_SECRET` is not supported as well, do I understand it right? Our use case is not use k8s secrets at all

@marekargalas You are correct that we don't support `AIRFLOW__CORE__SQL_ALCHEMY_CONN_SECRET`, as this would be overridden by the chart's generated `AIRFLOW__CORE__SQL_ALCHEMY_CONN` (which actually uses `AIRFLOW__CORE__SQL_ALCHEMY_CONN_CMD` internally to set it).

__NOTE:__ we might want to add the `_SECRET` configs to [`_helpers/validate-values.tpl`](https://github.com/airflow-helm/charts/blob/df99a33e3278de9f6935e747e8317794966ed979/charts/airflow/templates/_helpers/validate-values.tpl#L59-L62), so that the helm install will fail if users try and set `AIRFLOW__CORE__SQL_ALCHEMY_CONN_SECRET`. Do you want to raise a PR for that?
